Merge Request.

CorylC opened the pull request:
Merge Request between my main repository and the projects. New features include:
- Dana-8 endpoint
- nickel-strunz-10 endpoint
- countries endpoint
- option to name files
- finished functionality of previous endpoints

CorylC committed:
Countries update

added countriesRetriever class with id and page

CorylC committed:
Resubmit of fixed countries class

This should be heading to my personal repository.

CorylC committed:
Update README.md

Added info about country retriever

CorylC committed:
Dana-8 update

Added the DanaRetriever class with working functions for retrieving group and subgroups. Server side issue with the retrieve and id functions. Implementation should work when those are figured out

CorylC committed:
Merge branch 'Branch1' of https://github.com/CorylC/OpenMindatTest into Branch1

CorylC committed:
Merge pull request #1 from CorylC/Branch1

Main branch merge

CorylC committed:
Update README with dana8 info.md

Added a use case of the dana-8 retriever class

CorylC committed:
Nickel_strunz commit

Added nickel_strunz_10 class, also fixed some comments in countries retriever class and dana-8 class.

CorylC committed:
Update README.md

added nickel strunz info

CorylC committed:
Added get_file_path function

Added the get_file_path function to decrease lines and increase readability.

CorylC committed:
Merge branch 'main' of https://github.com/CorylC/OpenMindatTest

CorylC committed:
Fix up + file naming

Reworked sub endpoint options as well as adding an option to name files.

CorylC committed:
Small fixes before merge

raises error when inputting wrong file name, changed the way varieties work in geomaterialsIdRetriever.

CorylC committed:
Localities_age commit

Added the Localities_age endpoint.
Also changed the format of countries.py to cut down on conflicts.

CorylC committed:
Added More endpoints

Added the localities_status, localities_type, locgeoregion2, and photo count endpoint class files.

ChuBL commented on openmindat/countries.py:
```diff
@@ -0,0 +1,223 @@
+        self._params.clear()
+        self._params = {'format': 'json'}
+    
+    def id(self, ID :int):
```
I noticed that you applied type hinting to the parameter `ID`, which is good. However I would like to see consistency in programming since this is the only type hinting throughout the file, so it would be better if you applied it to all the function parameters. You can also omit it if you already have a detailed function description.

Another thing is you should keep an eye on your space usage: the space should be placed to the right of the colon instead of the left side.

ChuBL commented on openmindat/countries.py:
```diff
@@ -0,0 +1,223 @@
+            id (INT): The country id.
+
+        Returns:
+            self: The CountriesRetriver object.
```
You copied and pasted the function from the above class, which is why the function description mentioned the other class object instead of itself.

Copying and pasting are operations to which you should pay extra attention.

ChuBL commented on openmindat/geomaterials.py:
```diff
@@ -1,3 +1,4 @@
+from typing import Optional
```
This seems to be automatically generated by your VScode plugin, like autopep8, when you accidentally type the word "optional" outside of the comment zone, which misleads the compiler to recognize it as a function or something. 

Remove this line if you don't need it.

ChuBL commented on openmindat/locgeoregion2.py:
```diff
@@ -0,0 +1,126 @@
+
+
+class GeoRegionRetriever:
+    """
```
Mindat API Redoc: https://api.mindat.org/schema/redoc/#tag/locgeoregion2

I thought it would be a good idea to insert corresponding Redoc links in the class descriptions.

ChuBL commented on openmindat/localities_age.py:
```diff
@@ -0,0 +1,226 @@
+        
+        ma = mindat_api.MindatApi()
+        
+        if "page" in params:
```
It is good to know you noticed and handled the data structure difference between single-page retrieval and multiple ones.

ChuBL commented on openmindat/localities.py:
```diff
@@ -331,46 +331,144 @@ def updated_at(self, DATE_STR):
+        >>> lir.id(5)
+
+    Attributes:
+        _id (int): An int to store id parameter.
```
The initial underscore before the parameters of functions indicates that the object is a private target. However, this naming cannot prevent outside functions from calling the parameter. It is just a programming style for readability. So, please avoid using the initial underscore for non-private objects.

ChuBL commented on openmindat/minerals_ima.py:
```diff
@@ -269,25 +270,123 @@ def saveto(self, OUTDIR=''):
+        >>> midr.id(5)
+
+    Attributes:
+        _id (int): An int to store id parameter.
```
Remove the underscore.

ChuBL reviewed (changes requested):
I have left the comments for minor fixing. Please address the code with propagating issues and send the pull request again. Thank you.

CorylC committed:
Fix ups and locobject class

Changed the type hinting to type checking of the input on id methods. Fixed some comments, added redox links in comments, and removed some small problems.

ChuBL reviewed (approved):
Thank you for your timely modification!

CorylC commented:
You're welcome!
I did want to check in regarding the type hinting in the parameters. I figured it would be simpler to add type checking when it was needed in those id methods. If you would prefer I can change all of the parameters to type hinting.

Best,
      Cory.
________________________________
From: Jiyin Zhang ***@***.***>
Sent: Wednesday, March 6, 2024 10:12 PM
To: ChuBL/OpenMindat ***@***.***>
Cc: CorylC ***@***.***>; Author ***@***.***>
Subject: Re: [ChuBL/OpenMindat] Merge Request. (PR #2)


@ChuBL approved this pull request.

Thank you for your timely modification!

—
Reply to this email directly, view it on GitHub<https://github.com/ChuBL/OpenMindat/pull/2#pullrequestreview-1921519570>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BCW4I3KTLLZ5WVX77HIBN5DYXAANBAVCNFSM6AAAAABEBD46E2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMRRGUYTSNJXGA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

ChuBL commented:
Type hinting is a good programming style but optional. The developing progress for this project is almost done. If that's okay with you, let's keep it in mind for future Python projects. I am going to test the functions you added. I will sync your work to the [pypi](https://pypi.org/project/openmindat/) if they work well. Thank you for your hard work!

CorylC commented:
I forgot to mention that I couldn’t get the locobject endpoint to work, every id I tried gave me ""error": "Object not found"".

________________________________
From: Jiyin Zhang ***@***.***>
Sent: Wednesday, March 6, 2024 10:24:49 PM
To: ChuBL/OpenMindat ***@***.***>
Cc: CorylC ***@***.***>; Author ***@***.***>
Subject: Re: [ChuBL/OpenMindat] Merge Request. (PR #2)


Type hinting is a good programming style but optional. The developing progress for this project is almost done. If that's okay with you, let's keep it in mind for future Python projects. I am going to test the functions you added. I will sync your work to the pypi<https://pypi.org/project/openmindat/> if they work well. Thank you for your hard work!

—
Reply to this email directly, view it on GitHub<https://github.com/ChuBL/OpenMindat/pull/2#issuecomment-1982539850>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BCW4I3PI632SWZMLFFWTVTTYXAB3DAVCNFSM6AAAAABEBD46E2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBSGUZTSOBVGA>.
You are receiving this because you authored the thread.

ChuBL commented:
It's possible that the server is to blame. It is surprising that you didn't encounter the Server 500 Error during development, haha.

ChuBL commented on openmindat/countries.py:
```diff
@@ -0,0 +1,206 @@
+from . import mindat_api
+
+
+class CountriesRetriever:
```
Second revision:
1. The endpoint to this class is `countries_list`.  Therefore, please consider using another class name, e.g., "CountriesListRetriever".
2. The querying result for this class is incomplete: it only returns one page of items. Please consider programming a specific function for obtaining the countries in a for loop using your new function of getting the list results. This would be a tentative solution for users to obtain the entire countries list.

CorylC commented:
I have fixed these issues in the latest update to my personal repository.
________________________________
From: Jiyin Zhang ***@***.***>
Sent: Friday, March 22, 2024 4:31 PM
To: ChuBL/OpenMindat ***@***.***>
Cc: CorylC ***@***.***>; Author ***@***.***>
Subject: Re: [ChuBL/OpenMindat] Merge Request. (PR #2)


@ChuBL commented on this pull request.

________________________________

In openmindat/countries.py<https://github.com/ChuBL/OpenMindat/pull/2#discussion_r1536371236>:

> @@ -0,0 +1,206 @@
+from . import mindat_api
+
+
+class CountriesRetriever:


Second revision:

  1.  The endpoint to this class is countries_list. Therefore, please consider using another class name, e.g., "CountriesListRetriever".
  2.  The querying result for this class is incomplete: it only returns one page of items. Please consider programming a specific function for obtaining the countries in a for loop using your new function of getting the list results. This would be a tentative solution for users to obtain the entire countries list.

—
Reply to this email directly, view it on GitHub<https://github.com/ChuBL/OpenMindat/pull/2#pullrequestreview-1956044323>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BCW4I3ILWZCH7LWY2DAFAPTYZS5ORAVCNFSM6AAAAABEBD46E2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSNJWGA2DIMZSGM>.
You are receiving this because you authored the thread.Message ID: ***@***.***>
